.dockerignore

```diff
@@ -0,0 +1,39 @@
+# Docker build context for StarryOS
+#
+# Design:
+# - Ignore everything by default
+# - Then whitelist only what the Docker build actually needs
+# This keeps the file short and avoids duplicating .gitignore.
+
+# Ignore everything
+*
+
+# ---- Keep minimal metadata ----
+!.gitignore
+!.gitmodules
+
+# ---- Keep Docker-related files ----
+!docker/
+!docker/**
+
+# ---- Keep Rust toolchain and workspace manifests ----
+!rust-toolchain.toml
+!Cargo.toml
+!Cargo.lock
+
+# ---- Keep source crates ----
+!api/
+!api/**
+!core/
+!core/**
+!arceos/
+!arceos/**
+
+# ---- Keep top-level docs that are referenced in README / useful in container ----
+!README.md
+!LICENSE
+!NOTICE
+!docs/
+!docs/**
```

Comment:
Do we need so many files to build? I believe only `docker` folder and `rust-toolchain.toml` are necessary.